crates/holochain_sqlite/src/db.rs

```diff
@@ -237,6 +237,7 @@ impl<Kind: DbKindT + Send + Sync + 'static> DbWrite<Kind> {
         Self::open_with_sync_level(path_prefix, kind, DbSyncLevel::default())
     }
 
+    // TODO this isn't even used?
     pub async fn conn_write_permit(&self) -> PConnPermit {
         let g = self.acquire_writer_permit().await;
         PConnPermit(g)
```

Comment:
I vote to go ahead and remove.

```suggestion
```

Reply:
Absolutely agree but I'm keen to keep this PR focused on adding tests and changing the tested code as little as possible. I'm fine if that means deleting some of the tests on the next PR. I think it's easier to keep track of what's being changed and why it fixes something that way